arclightning_backend/src/main.rs

```diff
@@ -0,0 +1,108 @@
+extern crate futures;
+extern crate hyper;
+#[macro_use] extern crate serde_derive;
+#[macro_use] extern crate serde_json;
+extern crate toml;
+
+use futures::future;
+use hyper::rt::{Future, Stream};
+use hyper::service::service_fn;
+use hyper::{Body, Method, Request, Response, Server, StatusCode};
+use std::path::PathBuf;
+
+// after this concept is further understood, will switch to 'Either'
+type BoxFuture = Box<Future<Item = Response<Body>, Error = hyper::Error> + Send>;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+struct Game {
+    id: PathBuf,
+    name: PathBuf,
+    description: PathBuf,
+    genre: PathBuf, // TODO: make a vector
+    thumbnail_path: PathBuf,
+    exe_path: PathBuf,
+}
+
+fn router(request: Request<Body>) -> BoxFuture {
+
+    let mut games_list: Vec<Game> = Vec::new();
+
+    let mut response = Response::new(Body::empty());
+
+    match (request.method(), request.uri().path()) {
+        // TODO: get will send over games list
+        (&Method::GET, "/") => {
+            // TODO: convert a vector of struct "Game" into a single json:
+            *response.body_mut() = Body::from("games will go here");
+        }
+        // TODO: post will probably figure out which game to launch?
+        (&Method::POST, "/echo") => {
+            *response.body_mut() = request.into_body();
+        }
+        _ => {
+            *response.status_mut() = StatusCode::NOT_FOUND;
+        }
+    }
+
+    Box::new(future::ok(response))
+}
+
+fn main() {
+    let addr = ([127, 0, 0, 1], 3000).into();
+
+
+    let server = Server::bind(&addr)
+        .serve(|| service_fn(router))
+        .map_err(|e| eprintln!("server error: {}", e));
+
+
+    println!("Listening on http://{}", addr);
+    hyper::rt::run(server);
+}
+
+
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_read_toml() {
+
+        use std::fs::{File};
```

Comment:
Why is this by itself in curly brackets. I know rustfmt fixes this, so you might want to do a quick run of rustfmt to resolve this.